### PR TITLE
Video 9638 web implement chat

### PR DESCRIPTION
**Pull request by olipyskoty**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [VIDEO-9638](https://issues.corp.twilio.com/browse/VIDEO-9638)

### Description

This PR adds the audience chat to the web livestreaming app. Viewers can now view the chat and send messages.

### Before review
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [x] Verified locally with all effected platforms
* [x] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [x] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

**Review by timrozum — APPROVED** [review]
Room SID stuff looks good.

**Review by timmydoza — CHANGES_REQUESTED** [review]
I think this looks good, but some merge conflicts have appeared. @olipyskoty would you mind taking a look at these conflicts and then I'll take another look.

File: apps/web/src/components/Player/PlayerMenuBar/PlayerMenuBar.tsx [hunk 1]
```diff
@@ -63,12 +65,21 @@ export default function PlayerMenuBar({ roomName, disconnect }: { roomName?: str
...
   };
 
+  const toggleChatWindow = () => {
+    setIsChatWindowOpen(!isChatWindowOpen);
+    appDispatch({
+      type: 'set-is-participant-window-open',
+      isParticipantWindowOpen: false,
+    });
+  };
+
```
Comment (timmydoza) [inline apps/web/src/components/Player/PlayerMenuBar/PlayerMenuBar.tsx:82]: This is kinda funny - just that these two pieces of state that control two sidebars live in two different places. Not sure it's worth fixing now though.

File: apps/web/src/components/PreJoinScreens/PreJoinScreens.tsx [hunk 2]
```diff
@@ -44,48 +43,44 @@ export default function PreJoinScreens() {
...
-
         case 'viewer': {
           const { data } = await joinStreamAsViewer(appState.participantName, appState.eventName);
           syncConnect(data.token);
           await playerConnect(data.token);
           registerUserDocument(data.sync_object_names.user_document);
           registerSyncMaps(data.sync_object_names);
           await connectViewerToPlayer(appState.participantName, appState.eventName);
-          // chatConnect(data.token);
+          chatConnect(data.token, data.room_sid);
```
Comment (timmydoza) [inline apps/web/src/components/PreJoinScreens/PreJoinScreens.tsx:76]: Should this one be behind `if (data.chat_enabled) {` too?

**Review by timmydoza — APPROVED** [review]

**Pull request merged**